server/routes/prisonRegister/prisonRegisterController.ts

```diff
@@ -29,10 +29,19 @@ export default class PrisonRegisterController {
   }
 
   parseFilter(req: Request): AllPrisonsFilter {
+    let gendersFromQuery = CourtRegisterController.parseStringArrayFromQuery(req.query.genders as string[])
+    if (PrisonRegisterController.doNotFilterIfBothGendersInQuery(gendersFromQuery)) {
+      gendersFromQuery = undefined
```

Comment:
We don't need to provide a filter to the backend if the following scenario happens:
1. User clicks female checkbox in order to filter for male only
2. User clicks apply filter
3. User then re-clicks female checkbox

Hence we clear the array before constructing `AllPrisonsFilter`